l2lsp/methods/lsp_suggestEnv.go

```diff
@@ -0,0 +1,81 @@
+// lsp_suggestEnv.go
+package methods
+
+import (
+	"errors"
+	"path/filepath"
+	"strings"
+
+	l2envpackege "github.com/HexmosTech/lama2/l2env"
+	"github.com/HexmosTech/lama2/l2lsp/lsp_req"
+	"github.com/HexmosTech/lama2/l2lsp/lsp_res"
+	"github.com/rs/zerolog/log"
+)
+
+func getSearchQueryString(request lsp_req.JSONRPCRequest) string {
+	if request.Params.SearchQuery != nil {
+		return *request.Params.SearchQuery
+	}
+	return ""
+}
+
+func getRequestURI(request lsp_req.JSONRPCRequest) (string, int, error) {
+	if request.Params.TextDocument.Uri == nil {
+		return "", lsp_res.ErrInvalidURI, errors.New("URI cannot be empty. Ex: 'file:///path/to/workspace/myapi.l2'")
+	}
+	uri := *request.Params.TextDocument.Uri
+
+	// Handle local files
+	if strings.HasPrefix(uri, "file://") {
+		return uri[len("file://"):], 0, nil
+
+		// Handle remote files (example for SSH)
+	} else if strings.HasPrefix(uri, "ssh://") {
+		return uri[len("ssh://"):], 0, nil
```

Comment:
Shouldn't ssh and WSL also return an unsupported error?

Reply:
right made the change